x-pack/docs/en/security/troubleshooting.asciidoc

```diff
@@ -319,6 +320,76 @@ In this case, you must install the
 <<ciphers, JCE Unlimited Strength Jurisdiction Policy Files>>.
 --
 
+[[trb-security-kerberos]]
+=== Common Kerberos exceptions
+
+*Symptoms:*
+
+* User authentication fails due to either GSS negotiation failure 
+or a service login failure ( either on the server or in the {es} http client). 
+Some of the common exceptions are listed below with some tips to resolve them.
+
+*Resolution:*
+
+`Failure unspecified at GSS-API level (Mechanism level: Checksum failed)`::
++
+--
+
+When you see this error message on the HTTP client side, then it may be 
+related to an incorrect password.
+
+When you see this error message in the {es} server logs, then it may be 
+related to the {es} service keytab. The keytab file is present but it failed 
+to log in as the user. Please check the keytab expiry. Also check whether the 
+keytab contain up-to-date credentials; if not, replace them.
+
+You can use tools like `klist` or `ktab` to list principals inside 
+the keytab and validate them. You can use `kinit` to see if you can acquire 
+initial tickets using the keytab. Please check the tools and their documentation 
+in your Kerberos environment.
+
+Kerberos depends on proper hostname resolution, so please check your DNS infrastructure.
+Incorrect DNS setup, DNS SRV records or configuration for KDC servers in `krb5.conf` 
++can cause problems with hostname resolution.
+
+--
+
+`Failure unspecified at GSS-API level (Mechanism level: Request is a replay (34))`::
+
+`Failure unspecified at GSS-API level (Mechanism level: Clock skew too great (37))`::
++
+--
+
+To prevent replay attacks, Kerberos V5 sets a maximum tolerance for computer 
+clock synchronization and it is typically 5 minutes. Please check whether 
+the time on the machines within the domain is in sync.
+
+--
+
+As Kerberos logs are often cryptic in nature and many things can go wrong 
+as it depends on external services like DNS, time synchronization. You might 
+have to enable additional debug logs to root cause the issue.
```

Comment:
I don't think "root cause" is a verb 😈 

Maybe "... debug logs to determine the root cause of the issue."